src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-cluster-form/smb-cluster-form.component.ts

```diff
@@ -0,0 +1,262 @@
+    this.allClustering = Object.values(Clustering);
+  }
+
+  createForm() {
+    this.smbForm = this.formBuilder.group({
+      cluster_id: new FormControl('', {
+        validators: [Validators.required]
+      }),
+      auth_mode: [
+        '',
+        {
+          validators: [Validators.required]
+        }
+      ],
+      domain_settings: [''],
+      placement: [{}],
+      hosts: [[]],
+      label: [
+        null,
+        [
+          CdValidators.requiredIf({
+            placement: 'label'
+          })
+        ]
+      ],
+      custom_dns: new FormArray([]),
+      joinSources: new FormArray([]),
+      clustering: new UntypedFormControl(Clustering.Default)
+    });
+
+    this.orchService.status().subscribe((status) => {
+      this.hasOrchestrator = status.available;
+      this.smbForm.get('placement').setValue(this.hasOrchestrator ? 'hosts' : '');
+    });
+  }
+
+  multiSelector(event: any, field: 'label' | 'hosts') {
+    if (field === 'label') this.selectedLabels = event.map((label: any) => label.content);
+    else this.selectedHosts = event.map((host: any) => host.content);
+  }
+
+  onAuthModeChange(): void {
+    const authMode = this.smbForm.get('auth_mode').value;
+    const domainSettingsControl = this.smbForm.get('domain_settings');
+    const userGroupSettingsControl = this.smbForm.get('joinSources');
+
+    if (authMode === AuthMode.activeDirectory) {
+      const joinSources = this.smbForm.get('joinSources') as FormArray;
+      if (joinSources) {
+        joinSources.clear();
+      }
+      domainSettingsControl.setValidators(Validators.required);
+      userGroupSettingsControl.clearValidators();
+    } else if (authMode === AuthMode.User) {
+      const control = new FormControl('', Validators.required);
+      this.joinSources.push(control);
+      userGroupSettingsControl.setValidators(Validators.required);
+      domainSettingsControl.clearValidators();
+    }
+    domainSettingsControl.updateValueAndValidity();
+    userGroupSettingsControl.updateValueAndValidity();
+  }
+
+  submitAction() {
+    const requestModel = this.buildRequest();
+    let values = this.smbForm.getRawValue();
+    const serviceSpec: object = {
+      placement: {}
+    };
+    switch (values['placement']) {
+      case 'hosts':
+        if (values['hosts'].length > 0) {
+          serviceSpec['placement']['hosts'] = this.selectedHosts;
+        }
+        break;
+      case 'label':
+        serviceSpec['placement']['label'] = this.selectedLabels;
+        break;
+    }
+
+    const BASE_URL = 'smb/cluster';
+    const cluster_id = this.smbForm.get('cluster_id').value;
+    const taskUrl = `${BASE_URL}/${URLVerbs.CREATE}`;
+    this.taskWrapperService
+      .wrapTaskAroundCall({
+        task: new FinishedTask(taskUrl, { cluster_id }),
+        call: this.smbService.create(requestModel)
+      })
+      .subscribe({
+        complete: () => {
+          this.router.navigate([`cephfs/smb`]);
+        }
+      });
```

Comment:
```suggestion
      .subscribe({
        complete: () => {
          this.router.navigate([`cephfs/smb`]);
        },
        error() {
          component.smbForm.setErrors({ cdSubmitButton: true });
        },
      });
```

To avoid breaking the form on an error when submiting